client_test.go

```diff
@@ -0,0 +1,169 @@
+package xsolla
+
+import (
+	"fmt"
+	"github.com/joho/godotenv"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+)
+
+
+func newTestClient() *Client {
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Fatalf("Error loading .env file")
+	}
+	merchantID, _ := strconv.Atoi(os.Getenv("XsollaMerchantId"))
+	projectID, _ := strconv.Atoi(os.Getenv("XsollaProjectId"))
+	return &Client {
+		MerchantId:     merchantID,
+		MerchantSecret: os.Getenv("XsollaMerchantSecret"),
+		ProjectId:      projectID,
+		ProjectSecret:  os.Getenv("XsollaProjectSecret"),
+		Sandbox :       true,
+		Timeout :       100000000000,
+	}
+}
+
+
+
+func newTestToken() *Token {
+	UISettings := NewUISettings()
+	var utm M
+	var attr M
+	userData := NewUserData("user_2",
+		                    "john.smith@mail.com",
+		                    "",
+		                    utm,
+		                    attr,)
+	return &Token {
+		User : userData,
+		Settings : TokenSettings{
+			ProjectId: 122221,
+			UI : UISettings,
+			Mode : "",
+			ReturnURL : "",
+		},
+
+	}
+}
+
+func newTestUser() *User {
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Fatalf("Error loading .env file")
+	}
+	return &User{
+		Country : "Canada",
+		Email : "john.smith@mail.com",
+		Id : os.Getenv("testUserID"),
+		IP : "1.1.1.1",
+		Name : "John Smith",
+		Phone : "604-310-1010",
+		Zip : "90210",
+	}
+}
+
+func newTestAttribute() M{
+	return M {
+		"key" : "13",
+		"list_of_values" : M {},
+		"name" : M {
+			"en": "rating",
+		},
+		"type" : "int",
+	}
+}
+
+
+
+
+func TestClient_GetSubscriptionUserId(t *testing.T) {
+	testClient := newTestClient()
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Fatalf("Error loading .env file")
+	}
+	id := os.Getenv("testSubscriptionUserID")
+	userID, err := testClient.GetSubscriptionUserId(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if userID == "" {
+		t.Fatal("Returned userID was nill")
+	}
+}
+
+
+// No entry to test with ProjectID, will have to use 29054
+// Additionally, part of the payload returns a date however there
+// Seems to be an error when using json.Unmarshal() to parse the date
+func TestClient_GetSubscription(t *testing.T) {
+	testClient := newTestClient()
+	testClient.ProjectId = 29054
```

Comment:
I would use alternate ids instead of baking these in or simply use a different set of ids for testing purposes in your env vars.